source/common/network/active_raw_udp_listener_config.cc

```diff
@@ -0,0 +1,17 @@
+#include "common/network/active_raw_udp_listener_config.h"
```

Comment:
I'm following the factory naming conversion in envoy to name the factory class of a factory to be SomethingConfigFactory, e.g.  https://github.com/envoyproxy/envoy/blob/1e6c9fb44f8317c8a78e85100cf685c16c41033a/include/envoy/server/transport_socket_config.h#L103:7

ActiveUdpListenerFactory::createActiveUdpListener() is called where protobuf has already been converted into ListenerConfig object. So passing in protobuf is a bit weird there. That's why I used 2 layers of factory creation.

Reply:
ok, I do not like the example in your link too, but at least it is consistent with here.

I still think this should be simplified, e.g., by changing the ListenerConfig::udpListenerFactory() interface to ListenerConfig::udpListenerConfig(), get the (1-layer) ActiveUdpListenerFactory at addListener(), and use it to create the listener. Up to you whether you want to do it.

Reply:
Hmm, I see what you mean, but we would need to add a object form of typed protobuf UdpListenerConfig so that it can be carried in ListenerConfig. Not sure if it's more complicated or easier...

Reply:
I don't have a strong preference here, so will defer to both of you to decide.

Reply:
I added a TODO to investigating the approach of refactoring QuicConnection constructor.

Reply:
@danzh2010 : I meant to save the UdpListenerConfig message directly(not a "c++ form of it") in ListenerImpl, and return it from the ListenerConfig::udpListenerConfig() method. Again, I'll leave it up to you.

Reply:
I have see any protobuf stored in ListenerConfig, but only their class form. So I'm not sure if this is a good idea. I would like to keep the 2 layer of factories for now, and we can simplify the code later if necessary.